test/helpers/name.go

```diff
@@ -34,3 +45,45 @@ func ObjectPrefixForTest(t *testing.T) string {
 func ObjectNameForTest(t *testing.T) string {
 	return AppendRandomString(ObjectPrefixForTest(t))
 }
+
+// AppendRandomString will generate a random string that begins with prefix.
+// This is useful if you want to make sure that your tests can run at the same
+// time against the same environment without conflicting.
+// This method will use "-" as the separator between the prefix and
+// the random suffix.
+func AppendRandomString(prefix string) string {
+	suffix := make([]byte, randSuffixLen)
+
+	for i := range suffix {
+		suffix[i] = letterBytes[rand.Intn(len(letterBytes))]
+	}
+
+	return strings.Join([]string{prefix, string(suffix)}, string(sep))
+}
+
+// MakeK8sNamePrefix converts each chunk of non-alphanumeric character into a single dash
+// and also convert camelcase tokens into dash-delimited lowercase tokens.
+func MakeK8sNamePrefix(s string) string {
+	var sb strings.Builder
+	newToken := false
+	for _, c := range s {
+		if !(unicode.IsLetter(c) || unicode.IsNumber(c)) {
+			newToken = true
+			continue
+		}
+		if sb.Len() > 0 && (newToken || unicode.IsUpper(c)) {
+			sb.WriteRune(sep)
+		}
+		sb.WriteRune(unicode.ToLower(c))
+		newToken = false
+	}
+	return sb.String()
+}
+
+// GetBaseFuncName returns the baseFuncName parsed from the fullFuncName.
+// eg. test/e2e.TestMain will return TestMain.
+func GetBaseFuncName(fullFuncName string) string {
+	baseFuncName := fullFuncName[strings.LastIndex(fullFuncName, "/")+1:]
```

Comment:
are we guaranteed a slash here?

Reply:
It's not guaranteed, so I modified it a bit to make the logic clearer, does it look better to you?

Reply:
It's fine.